modules/libcom/src/osi/os/posix/osdThread.c

```diff
@@ -427,6 +440,11 @@ static void epicsThreadInit(void)
     static pthread_once_t once_control = PTHREAD_ONCE_INIT;
     int status = pthread_once(&once_control,once);
     checkStatusQuit(status,"pthread_once","epicsThreadInit");
+
+    if(epicsAtomicGetIntT(&childAfterFork)==1 &&  epicsAtomicCmpAndSwapIntT(&childAfterFork, 1, 2)==1) {
```

Comment:
The call to `epicsAtomicGetIntT()` can't be necessary here (if it were there would be a race in the time between the two atomic calls). If the value isn't `forkDone` at that time the call to `epicsAtomicCmpAndSwapIntT()` shouldn't succeed, which gives the same outcome with fewer atomic operations.

Reply:
Correction it's not actually a race, but the warning could get missed.

Reply:
Nope, not necessary.  This is an optimization (perhaps premature) to avoid the cost of a more expensive atomic CAS.  C++ `std::atomic` forced me into this pattern.  The first argument of [`compare_exchange_strong()`](https://en.cppreference.com/w/cpp/atomic/atomic/compare_exchange) is a non-const reference.  I think I see, and agree, with the logic though.